src/settings/SettingsAddress.js

```diff
@@ -58,18 +57,7 @@ export default class SettingsAddress extends React.Component {
 			address_postcode: state.postcode,
 			phone: state.tel,
 		};
-		LiveData.api.setAccountSettings(req).then(
-			response => {
-				if (response.set_settings === 1) {
-					this.actions.updateSettingFields(req);
-				} else {
-					SettingsAddress.handleUpdateError(response);
-				}
-			},
-			response => {
-				SettingsAddress.handleUpdateError(response);
-			}
-		);
+		this.props.actions.attemptUpdateSettings(req, SettingsAddress.handleUpdateError);
```

Comment:
sure ~ was trying to have some name for things that I expect it to fail ~ 
most actions should not fail, so how should we deal with those that might fail?

AFAIK, actions does not return meaningful result by default, which makes it harder to use for read purpose api call 

any thoughts?